### PR TITLE
Document and Banking application modified with scalardl-java-client-sdk 2.0.4

### PR DESCRIPTION
**Pull request by scalar-boney**
A simple bank account application document updated with scalardl-java-client-sdk 2.0.4
A simple bank account application source code updated with scalardl-java-client-sdk 2.0.4

### CODE REVIEW

File: docs/applications/simple-bank-account/README.md [hunk 1]
```diff
@@ -207,16 +207,15 @@ try (ClientService clientService = injector.getInstance(ClientService.class)) {
...
 ```java
 JsonObject argument = Json.createObjectBuilder().add("id", "010-123456789").build();
-ContractExecutionResponse response = clientService.executeContract("create-account", argument);
+ContractExecutionResult response = clientService.executeContract("create-account", argument);
 ```
 
 will execute the `CreateAccount` contract with argument `{"id": "010-123456789"}`, as we did above. Note that we call the contract using the supplied id `create-account` that we chose when registering the contract.
 
-The result of executing the contract is a `ContractExecutionResponse`. It contains, among other things, a status code, error message, and result, each of which can be obtained respectively as
+The result of executing the contract is a `ContractExecutionResult`. It contains, proofs and result, each of which can be obtained respectively as
```
Comment (feeblefakie) [inline docs/applications/simple-bank-account/README.md]:
```suggestion
The result of executing the contract is a `ContractExecutionResult`. It contains, result and proofs, each of which can be obtained respectively as
```

File: docs/applications/simple-bank-account/README.md [hunk 2]
```diff
@@ -207,16 +207,15 @@ try (ClientService clientService = injector.getInstance(ClientService.class)) {
 
 ```java
 JsonObject argument = Json.createObjectBuilder().add("id", "010-123456789").build();
-ContractExecutionResponse response = clientService.executeContract("create-account", argument);
+ContractExecutionResult response = clientService.executeContract("create-account", argument);
```
Comment (feeblefakie) [inline docs/applications/simple-bank-account/README.md]:
```suggestion
ContractExecutionResult result = clientService.executeContract("create-account", argument);
```

File: docs/applications/simple-bank-account/README.md [hunk 3]
```diff
@@ -207,16 +207,15 @@ try (ClientService clientService = injector.getInstance(ClientService.class)) {
...
 
 will execute the `CreateAccount` contract with argument `{"id": "010-123456789"}`, as we did above. Note that we call the contract using the supplied id `create-account` that we chose when registering the contract.
 
-The result of executing the contract is a `ContractExecutionResponse`. It contains, among other things, a status code, error message, and result, each of which can be obtained respectively as
+The result of executing the contract is a `ContractExecutionResult`. It contains, proofs and result, each of which can be obtained respectively as
 
 ```java
-response.getStatus();
-response.getMessage();
+response.getProofs()
```
Comment (feeblefakie) [inline docs/applications/simple-bank-account/README.md]:
```suggestion
result.getProofs()
```

File: docs/applications/simple-bank-account/README.md [hunk 4]
```diff
@@ -207,16 +207,15 @@ try (ClientService clientService = injector.getInstance(ClientService.class)) {
...
 response.getResult();
```
Comment (feeblefakie) [inline docs/applications/simple-bank-account/README.md]:
```suggestion
result.getResult();
```

Comment (scalar-boney) [reply docs/applications/simple-bank-account/README.md]: Thank you for review comments.
Changes updated in 07087f3cf6c0ba8d8349ec1e9d18e9d8cc8f0a61

File: docs/applications/simple-bank-account/README.md (hunk 3 above)
Comment (scalar-boney) [reply docs/applications/simple-bank-account/README.md]: Thank you for review comments.
Changes updated in 07087f3

File: docs/applications/simple-bank-account/README.md (hunk 2 above)
Comment (scalar-boney) [reply docs/applications/simple-bank-account/README.md]: Thank you for review comments.
Changes updated in 07087f3

File: docs/applications/simple-bank-account/README.md (hunk 1 above)
Comment (scalar-boney) [reply docs/applications/simple-bank-account/README.md]: Thank you for review comments.
Changes updated in 07087f3

**Comment by feeblefakie** [comment]
@scalar-boney Did you check if it works properly ?

**Comment by feeblefakie** [comment]
@scalar-boney Also please apply google-java-format if not.

File: docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/service/AccountService.java [hunk 5]
```diff
@@ -68,21 +68,21 @@
...
-
-      if (response.getStatus() == StatusCode.OK.get()) {
-        return ResponseEntity.ok(response.getResult());
-      } else {
-        return ResponseEntity.badRequest()
-            .body("status: " + response.getStatus() + ", message: " + response.getMessage());
-      }
+      ContractExecutionResult result = f.apply(json);
+
+      return ResponseEntity.ok(result.getResult().toString());
```
Comment (feeblefakie) [inline docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/service/AccountService.java]: It is probably not correct.
First, `result.getResult()` returns `Optional` so it shouldn't do `toString()` directly.
Second, is it OK to set null to `ResponseEntity.ok` ? If it is not, it should return `ResponseEntity.ok()` when it is empty.
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/ResponseEntity.html#ok-T-

File: docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/service/AccountService.java [hunk 6]
```diff
@@ -68,21 +68,21 @@
...
+    } catch (ClientException e) {
+      return ResponseEntity.badRequest()
```
Comment (feeblefakie) [inline docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/service/AccountService.java]: It is not you, but I don't think it's good to return `BAD REQUEST(400)` here.
Can we return 500 instead (, which is also not correct, but better than 400) ?

**Comment by scalar-boney** [comment]
> @scalar-boney Did you check if it works properly ?

I have already verified it with docker.

**Comment by scalar-boney** [comment]
> @scalar-boney Also please apply google-java-format if not.

google-java-format applied.

File: docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/service/AccountService.java (hunk 5 above)
Comment (scalar-boney) [reply docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/service/AccountService.java]: Thank you for review comment.
Changes updated in d02b87aac798ec1dca93c22bc09f02a3f63d8161
```
return ResponseEntity
              .ok(result.getResult().isPresent() ? result.getResult().get().toString() : null);
```

File: docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/service/AccountService.java (hunk 6 above)
Comment (scalar-boney) [reply docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/service/AccountService.java]: Thank you for review comment.
Changes updated in d02b87a

File: docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/service/AccountService.java [hunk 7]
```diff
@@ -68,21 +68,22 @@
...
-      if (response.getStatus() == StatusCode.OK.get()) {
-        return ResponseEntity.ok(response.getResult());
-      } else {
-        return ResponseEntity.badRequest()
-            .body("status: " + response.getStatus() + ", message: " + response.getMessage());
-      }
+      ContractExecutionResult result = f.apply(json);
+
+      return ResponseEntity
+              .ok(result.getResult().isPresent() ? result.getResult().get().toString() : null);
```
Comment (feeblefakie) [inline docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/service/AccountService.java]:
```suggestion
              .ok(result.getResult().get());
```

File: docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/repository/AccountRepository.java [hunk 8]
```diff
@@ -33,46 +33,42 @@ public AccountRepository(@Value("${client.properties.path}") String properties)
...
+  private void logResponse(String header, ContractExecutionResult result) {
     logger.info(
         header
             + ": ("
-            + response.getStatus()
-            + ", "
-            + response.getMessage()
-            + ", "
-            + response.getResult()
+            + (result.getResult().isPresent() ? result.getResult().get() : "")
```
Comment (feeblefakie) [inline docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/repository/AccountRepository.java]:
```suggestion
            + (result.getResult().orElse(""))
```

File: docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/service/AccountService.java (hunk 7 above)
Comment (scalar-boney) [reply docs/applications/simple-bank-account/src/main/java/com/scalar/application/bankaccount/service/AccountService.java]: Thank you for review comments.
`result.getResult()`  maybe null, so checking is required.

So code modified like this based on discussion
```
return ResponseEntity
              .ok(result.getResult().isPresent() ? result.getResult().get().toString() : "{}");
```

**Pull request merged**